ddtrace/span.py

```diff
@@ -149,7 +148,10 @@ def __init__(
         self.trace_id = trace_id or _rand.rand64bits()  # type: int
         self.span_id = span_id or _rand.rand64bits()  # type: int
         self.parent_id = parent_id  # type: Optional[int]
-        self.tracer = tracer  # type: Optional[Tracer]
+        self._tracer = None  # type: Optional[Tracer]
+        if tracer is not None:
+            # log deprecation warning if tracer argument is not None
```

Comment:
should there be a log message generated here?

Reply:
so maybe delete the comment or mention that log happens with the call to the setter?